crates/driver/src/infra/notify/mod.rs

```diff
@@ -27,29 +27,13 @@ pub fn scoring_failed(
     solver: &Solver,
     auction_id: Option<auction::Id>,
     solution_id: &solution::Id,
-    err: &score::Error,
+    err: &solution::error::Scoring,
 ) {
     let notification = match err {
-        score::Error::ZeroScore => {
-            notification::Kind::ScoringFailed(notification::ScoreKind::ZeroScore)
+        solution::error::Scoring::Solution(solution::error::Solution::InvalidClearingPrices) => {
+            notification::Kind::ScoringFailed(ScoreKind::InvalidClearingPrices)
         }
-        score::Error::ScoreHigherThanQuality(score, quality) => notification::Kind::ScoringFailed(
-            notification::ScoreKind::ScoreHigherThanQuality(*score, *quality),
-        ),
-        score::Error::RiskAdjusted(score::risk::Error::SuccessProbabilityOutOfRange(
-            success_probability,
-        )) => notification::Kind::ScoringFailed(
-            notification::ScoreKind::SuccessProbabilityOutOfRange(*success_probability),
-        ),
-        score::Error::RiskAdjusted(score::risk::Error::ObjectiveValueNonPositive(
-            quality,
-            gas_cost,
-        )) => notification::Kind::ScoringFailed(
-            notification::ScoreKind::ObjectiveValueNonPositive(*quality, *gas_cost),
-        ),
-        score::Error::RiskAdjusted(score::risk::Error::Boundary(_)) => return,
-        score::Error::Boundary(_) => return,
-        score::Error::Scoring(_) => return, // TODO: should we notify?
+        _ => return,
```

Comment:
Can we keep listing all other variants explicitly? That would help to avoid mistakes in the future if a new variant is added, for example.

Reply:
I simplified the error enum and listed all of them. Can be additionally cleaned up as part of
https://github.com/cowprotocol/services/issues/2597

Reply:
Actually forgot to commit 😶‍🌫️  will do it in a follow up PR